testing/test-utils/src/lib/utils/test-folder-setup.ts

```diff
@@ -36,3 +37,88 @@ export async function teardownTestFolder(dirName: string) {
     );
   }
 }
+
+/**
+ * File names that need to be restored by removing the "_" prefix.
+ * These files are prefixed with "_" in mock fixtures to avoid Nx detection.
+ */
+export const NX_IGNORED_FILES_TO_RESTORE = [
+  '_package.json',
+  '_nx.json',
+  '_project.json',
+] as const;
+
+/**
+ * File names that need to be prefixed with "_" to avoid Nx detection.
+ */
+export const NX_IGNORED_FILES_TO_PREFIX = [
+  'package.json',
+  'nx.json',
+  'project.json',
+] as const;
+
+/**
+ * Recursively renames specific files by removing the "_" prefix.
+ * This is needed because mock fixtures have "_" prefix to avoid Nx detection,
+ * but tests need the original filenames.
+ *
+ * @param dir - Directory to process recursively
+ * @param fileNames - Array of file names to restore (e.g., ['_package.json', '_nx.json', '_project.json'])
+ */
+export async function restoreRenamedFiles(
+  dir: string,
+  fileNames: readonly string[],
+): Promise<void> {
+  try {
+    const entries = await readdir(dir, { withFileTypes: true });
+    for (const entry of entries) {
+      const fullPath = path.join(dir, entry.name);
+      if (entry.isDirectory()) {
+        await restoreRenamedFiles(fullPath, fileNames);
+      } else if (entry.isFile() && fileNames.includes(entry.name)) {
+        const newName = entry.name.slice(1); // Remove leading "_"
+        const newPath = path.join(dir, newName);
+        try {
+          await rename(fullPath, newPath);
+        } catch (error) {
+          // Ignore errors if file doesn't exist or can't be renamed
+        }
+      }
+    }
+  } catch (error) {
+    // Ignore errors if directory doesn't exist
+  }
```

Comment:
Are you expecting these to fail? The paths are from `readdir`, so I would expect it's safe to traverse and rename any of these files. If there was an error, I'd be interested to know what it was, rather than silencing it.

Reply:
Very good point!! I wanted to open a pr showing the error and suggested solution but did not manage to understand the issue yesterday. Expect comments today and a PR in the next days.